devito/types/sparse.py

```diff
@@ -826,10 +838,21 @@ def _dist_gather(self, data):
         mpitype = MPI._typedict[np.dtype(self.dtype).char]
         comm.Alltoallv([data, rcount, rdisp, mpitype],
                        [gathered, scount, sdisp, mpitype])
-        data = gathered
         # Unpack data values so that they follow the expected storage layout
-        data = np.ascontiguousarray(np.transpose(data, self._dist_reorder_mask))
-        self._data[:] = data[self._dist_gather_mask]
+        gathered = np.ascontiguousarray(np.transpose(gathered, self._dist_reorder_mask))
+        self._data[:] = gathered[self._dist_gather_mask]
+
+        if coords is not None:
```

Comment:
Empty array is no an issue this is not for that case, this is because it can actually be None for example with `norm(rec)`  or `mmin/mmax(rec)` where the coordinates are not in args or kwargs and therefore `args[p.coordinates.name]` is None.

Reply:
ah ! fair point